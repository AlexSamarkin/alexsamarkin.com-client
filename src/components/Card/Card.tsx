import React from "react";

export interface CardProps {
  icon: string;
  title: string;
  children:
    | React.ReactNode
    | React.ReactNode[]
    | React.ReactNodeArray
    | React.ReactChildren
    | string;
}

export const Card: React.FC<CardProps> = (props) => {
  const { icon, title, children } = props;
  return (
    <div className="case-item box box__second">
      <img className="case-item__icon" src={icon} alt={title} />
      <div>
        <h3 className="title title--h5">{title}</h3>
        <p className="case-item__caption">{children}</p>
      </div>
    </div>
  );
};

export default Card;
