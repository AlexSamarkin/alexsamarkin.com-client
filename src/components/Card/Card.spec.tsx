import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card, { CardProps } from './Card';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps: CardProps = {
    icon: 'icon.svg',
    title: 'test',
    children: <div>123</div>,
};

describe('Card component', () => {
    test('renders', () => {
        const wrapper = shallow(<Card {...mockedProps} />);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
