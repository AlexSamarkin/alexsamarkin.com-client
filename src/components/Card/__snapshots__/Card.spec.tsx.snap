// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders 1`] = `ShallowWrapper {}`;
